### PR TITLE
[docs] add section on debugging performance issues

### PR DESCRIPTION
**Pull request by odow**

### CODE REVIEW

File: docs/src/tutorials/getting_started/debugging.jl [hunk 1]
```diff
@@ -355,3 +355,78 @@ for var in all_variables(model)
...
+
+# The strategy to debug JuMP models that have performance problems depends on
+# how long your model takes to build.
+
+# As a first step, encapsulate everything you want to debug into a single
+# function `foo`, so you can run it with `@time foo()`.
+
+# Can you run `@time foo()` in seconds to minutes?
+
+# If the answer is "yes", then you can use [ProfileView.jl](https://github.com/timholy/ProfileView.jl)
```
Comment (odow) [inline docs/src/tutorials/getting_started/debugging.jl:399]:
```suggestion
# If the answer is "yes," then you can use [ProfileView.jl](https://github.com/timholy/ProfileView.jl)
```

File: docs/src/tutorials/getting_started/debugging.jl [hunk 2]
```diff
@@ -355,3 +355,78 @@ for var in all_variables(model)
...
+
+# If a wide bar points to code inside JuMP or a related Julia package, please
+# open an issue on GitHub or the [community forum](https://jump.dev/forum).
+
+# If `@time foo()` takes longer than a few minutes to run, then either make the
+# problem smaller by using a smaller dataset, or do the following.
+#
+# 1. Comment out everything in the function, then, line by line (or block by
+#    block):
+# 2. Uncomment some code and re-run `@time foo()`
```
Comment (odow) [inline docs/src/tutorials/getting_started/debugging.jl:427]:
```suggestion
# 2. Un-comment some code and re-run `@time foo()`
```

**Comment by blegat** [comment]
I also find it helpful to just call `attach_optimizer` but not `optimize!` to see which part is copying the model

File: docs/src/tutorials/getting_started/debugging.jl [hunk 3]
```diff
@@ -355,3 +355,78 @@ for var in all_variables(model)
...
+# If `@time foo()` takes longer than a few minutes to run, then either make the
+# problem smaller by using a smaller dataset, or do the following.
+#
+# 1. Comment out everything in the function, then, line by line (or block by
+#    block):
+# 2. Un-comment some code and re-run `@time foo()`
+# 3. If the time increases by a lot (from seconds or minutes to hours), look
+#    for $O(N^2)$ or worse scaling behavior. Is there a better way to write the
+#    code that you are trying to execute?
+# 4. If the time increases by more than expected, but it still takes second or
```
Comment (odow) [inline docs/src/tutorials/getting_started/debugging.jl:431]:
```suggestion
# 4. If the time increases by more than expected, but it still takes seconds or
```

File: docs/src/tutorials/getting_started/debugging.jl [hunk 4]
```diff
@@ -355,3 +355,78 @@ for var in all_variables(model)
...
+# bars take more time. The bars are stacked so that the `foo()` call is on the
+# bottom, and subsequent calls within `foo` are stacked on top.
+
+# Reading a flamegraph can take some experience, but if you click on a bar it
+# will print the line number to the REPL. Hunt around until you find the widest
+# bar that points to a line of code that you have written, then ask yourself if
+# it makes sense for this line to be the bottleneck.
+
+# If a wide bar points to code inside JuMP or a related Julia package, please
+# open an issue on GitHub or the [community forum](https://jump.dev/forum).
```
Comment (odow) [inline docs/src/tutorials/getting_started/debugging.jl:420]:
```suggestion
# open an issue on GitHub or post on the [community forum](https://jump.dev/forum).
```

**Comment by odow** [comment]
I chose not to mention `attach_optimizer`. It's a bit too complicated for a common user question in a getting started tutorial.

**Pull request merged**